[Woo POS] [Cash & Receipts] Disable cash back button while loading

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #14908
<!-- Id number of the GitHub issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
We disable and change the style of the back button while we're collecting cash payment.

## Testing information
- Enable `.acceptCashForPointOfSale`
- Process a POS order and select cash
- After tapping "Mark as complete", observe how the navigation back button to the top left cannot be tapped, and the style is updated to reflect so (for easiness of testing you may want to add a `try await Task.sleep(nanoseconds: 10_000_000_000)` to `markComplete()`

![Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2025-01-20 at 13 24 57](https://github.com/user-attachments/assets/c5ae7c83-3874-4647-bbb2-af58ac5acd32)


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
Disable back button while loading

joshheald commented on WooCommerce/Classes/POS/Presentation/PointOfSaleCollectCashView.swift:
```diff
@@ -35,20 +35,19 @@ struct PointOfSaleCollectCashView: View {
                         Image(systemName: "chevron.left")
                             .font(.posTitleRegular)
                             .bold()
-                            .foregroundColor(.primary)
                         VStack(alignment: .leading, spacing: Constants.navigationButtonSpacing) {
                             Text(Localization.backNavigationTitle)
                                 .font(.posTitleRegular)
                                 .bold()
-                                .foregroundColor(.primary)
 
                             Text(formattedOrderTotal)
                                 .font(.posBodyRegular)
-                                .foregroundColor(.primary)
```
Nice removal of the clutter

joshheald reviewed (approved):
Works well, :shipit: 

I tested the happy path, error flow, multiple transactions, and switching between card and cash for the same transaction.

iamgabrielma committed:
Merge branch 'trunk' into issue/14908-disable-back-button-while-processing-cash

# Conflicts:
#	WooCommerce/Classes/POS/Presentation/PointOfSaleCollectCashView.swift

iamgabrielma committed:
Remove .foreground color duplication

These two were restored accidentally when addressing the merge conflict from the previous commit. We’re removing them again.
